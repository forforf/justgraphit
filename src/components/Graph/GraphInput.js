import React, {useState} from 'react';
import {Button, TextField} from '@mui/material';


function GraphInput({graphName, addNewNumber}) {

  // state
  const [nameInput, setNameInput] = useState(graphName)
  const [numberInput, setNumberInput] = useState('')

  const isNumeric = val => !isNaN(val) // If it's not, not a number(nan), then it's a number

  const handleSubmit = event => {
    if(isNumeric(numberInput)) {
      if(nameInput && nameInput.length) {
        addNewNumber(nameInput, numberInput)
      }
    }
    event.preventDefault();
    setNumberInput('');
  }

  const handleEnterKey = event => {
    if(event.key === "Enter"){
      handleSubmit(event);
    }
  };

  return (
    <div className="GraphInput">
      <div className="GraphInput-name">
        <h3>{nameInput}</h3>
      </div>
      <form onKeyPress={handleEnterKey}>
        <TextField
          placeholder="Name your graph"
          value={nameInput}
          onChange={event => setNameInput(event.target.value)}
        /><br />
        <TextField
          type="number"
          placeholder="Enter number to graph"
          value={numberInput}
          onChange={event => setNumberInput(event.target.value)}
        /><br />
        <Button variant="contained" onClick={handleSubmit}>Save
        </Button>
      </form>

    </div>
  );
}

export default GraphInput;
