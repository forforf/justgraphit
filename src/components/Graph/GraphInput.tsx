import React, { useState } from 'react';
import { Button, TextField } from '@mui/material';
import { GraphEntryHandler, GraphName } from '../../JustGraphitTypes';

export type GraphInputProps = {
  graphName: GraphName;
  addNewNumber: GraphEntryHandler;
};

function GraphInput({ graphName, addNewNumber }: GraphInputProps): JSX.Element {
  // state
  const [nameInput, setNameInput] = useState(graphName);
  const [numberInput, setNumberInput] = useState('');

  const isNumeric = (val: string): boolean => !isNaN(+val); // If it's not, not a number(nan), then it's a number

  const handleSubmit = (event: React.FormEvent) => {
    if (isNumeric(numberInput)) {
      if (nameInput && nameInput.length) {
        addNewNumber(nameInput, numberInput);
      }
    }
    event.preventDefault();
    setNumberInput('');
  };

  const handleEnterKey = (event: React.KeyboardEvent) => {
    if (event.key === 'Enter') {
      handleSubmit(event as React.FormEvent);
    }
  };

  return (
    <div className="GraphInput">
      <div className="GraphInput-name">
        <h3>{nameInput}</h3>
      </div>
      <form>
        <TextField
          placeholder="Name your graph"
          value={nameInput}
          onChange={(event) => setNameInput(event.target.value)}
          onKeyPress={handleEnterKey}
        />
        <br />
        <TextField
          type="number"
          placeholder="Enter number to graph"
          value={numberInput}
          onChange={(event) => setNumberInput(event.target.value)}
          onKeyPress={handleEnterKey}
        />
        <br />
        <Button
          variant="contained"
          onClick={handleSubmit}
          onKeyPress={handleEnterKey}
        >
          Save
        </Button>
      </form>
    </div>
  );
}

export default GraphInput;
