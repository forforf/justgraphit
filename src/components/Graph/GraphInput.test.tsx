import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';
import GraphInput, { GraphInputProps } from './GraphInput';

let graphInputProps: GraphInputProps;
beforeEach(() => {
  graphInputProps = {
    graphName: 'graph name',
    addNewNumber: jest.fn(),
  };
});

test('deep snapshot', () => {
  const component = renderer.create(<GraphInput {...graphInputProps} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('shallow snapshot', () => {
  const renderer = ShallowRenderer.createRenderer();
  renderer.render(<GraphInput {...graphInputProps} />);
  const result = renderer.getRenderOutput();
  expect(result).toMatchSnapshot();
});
