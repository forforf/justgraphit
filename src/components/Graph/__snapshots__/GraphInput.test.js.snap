// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`deep snapshot 1`] = `
<div
  className="GraphInput"
>
  <div
    className="GraphInput-name"
  >
    <h3>
      graph name
    </h3>
  </div>
  <form
    onKeyPress={[Function]}
  >
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Name your graph"
          required={false}
          type="text"
          value="graph name"
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1a1ihhj"
          >
            <span
              className="notranslate"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <br />
    <div
      className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
    >
      <div
        className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        onClick={[Function]}
      >
        <input
          aria-invalid={false}
          autoFocus={false}
          className="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
          disabled={false}
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Enter number to graph"
          required={false}
          type="number"
          value=""
        />
        <fieldset
          aria-hidden={true}
          className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="css-1a1ihhj"
          >
            <span
              className="notranslate"
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <br />
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Save
    </button>
  </form>
</div>
`;

exports[`shallow snapshot 1`] = `
<div
  className="GraphInput"
>
  <div
    className="GraphInput-name"
  >
    <h3>
      graph name
    </h3>
  </div>
  <form
    onKeyPress={[Function]}
  >
    <ForwardRef(TextField)
      onChange={[Function]}
      placeholder="Name your graph"
      value="graph name"
    />
    <br />
    <ForwardRef(TextField)
      onChange={[Function]}
      placeholder="Enter number to graph"
      type="number"
      value=""
    />
    <br />
    <ForwardRef(Button)
      onClick={[Function]}
      variant="contained"
    >
      Save
    </ForwardRef(Button)>
  </form>
</div>
`;

exports[`useEffect 1`] = `
<div>
  <div
    class="GraphInput"
  >
    <div
      class="GraphInput-name"
    >
      <h3>
        graph name
      </h3>
    </div>
    <form>
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            placeholder="Name your graph"
            type="text"
            value="graph name"
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1a1ihhj"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <br />
      <div
        class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
            placeholder="Enter number to graph"
            type="number"
            value=""
          />
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-1a1ihhj"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
      <br />
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Save
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>
</div>
`;
